### PR TITLE
Firefly-1252: spacial missing

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1252](https://jira.ipac.caltech.edu/browse/FIREFLY-1252): fixed issues with there are not spacial columns

  - changed the processing of tap.additional.services to better reflex the config orider
  - added hide property to tap.additional.services
  - temporal column check

#### Testing
 - https://fireflydev.ipac.caltech.edu/firefly-1252-spacial-missing/firefly
 - test according to the check
 - `tap.additional.services` and temporal column check changes are hard to test outside the Rubin environment

### CODE REVIEW

**Review by loitly — APPROVED** [review]
The reported bug is fixed.  Changes look good.  Btw, I added a commit to fix a regression bug when TAP results is not at the first item in the /results list.

File: src/firefly/js/ui/tap/SpatialSearch.jsx [hunk 1]
```diff
@@ -717,6 +718,9 @@ function makeSpatialConstraints(columnsModel, obsCoreEnabled, fldObj, uploadInfo
         const cenLat= cenLatField?.value;
         errList.checkForError(cenLonField);
         errList.checkForError(cenLatField);
+        if (!cenLon && !cenLat) errList.addError('Lon and Lat columns are not set');
```
Comment (gpdf) [inline src/firefly/js/ui/tap/SpatialSearch.jsx:721]: I'm not seeing this message displayed on the current data-int deployment; instead, I'm still seeing "Can not be empty.  Please provide value or expression"

File: src/firefly/js/ui/tap/TapUtil.js [hunk 2]
```diff
@@ -287,13 +287,20 @@ export const getAsEntryForTableName= (tableName) => tableName?.[0] ?? 'x';
...
 
-    const modifiedOriginal= tapServices.map( (t) => {
-        const match= additional.find( (a) => a.label===t.label);
-        return match ? {...t,...match} : t;
-    });
-    const trulyAdditional= additional.filter( (a) => !tapServices.find( (t) => t.label===a.label) );
-
-    return [...trulyAdditional,...modifiedOriginal];
+    const mergeAdditional= additional.map( (a) => {
+        if (a.hide) return false;
```
Comment (gpdf) [inline src/firefly/js/ui/tap/TapUtil.js:291]: I'm not sure how to use "hide" in the JSON, and it looks like I'll want to in order to provide the cleanest user experience.  Ca you provide an example?

**Review by gpdf — COMMENTED** [review]
One thing isn't working, on the data-int deployment, as the code above suggests it should: the missing-column-name error message.

**Pull request merged**